File: Data Structures & Algorithms/generate-parentheses/submission-3.py
class Solution:
    def generateParenthesis(self, n: int) -> List[str]:
        stack=[]
        def backTrack(left,right,res):
            if left==right==n:
                stack.append("".join(res))

            if left<n:
                res.append("(")
                backTrack(left+1,right,res)
                res.pop()
            
            if right<left:
                res.append(")")
                backTrack(left,right+1,res)
                res.pop()

        backTrack(0,0,[])
        return stack

File: Data Structures & Algorithms/daily-temperatures/submission-3.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int[]res=new int[temperatures.length];
        Arrays.fill(res,0);
        Deque<int[]> stack = new ArrayDeque<>();
        for(int i=0;i<temperatures.length;i++){
            while(!stack.isEmpty() && stack.peek()[0]<temperatures[i]){
                int[]entry=stack.pop();
                res[entry[1]]=i-entry[1];
            }
            
            stack.push(new int[]{temperatures[i],i});
            
        }
        return res;
    }
}
